### DISCUSSION TITLE
Check for `.displayName` property on actions / guards / services for display purposes in the inspector

### DISCUSSION THREAD

**Discussion by VanTanev** [statelyai/xstate#2247] [Ideas]
Currently, externally defined assign's name cannot be inferred:
```typescript
// shows up as "assign" in inspector
const setCount = assign({ count: (_, e) => e.count })
```

The inspector should check for a `displayName property and use that when available.

```typescript
const setCount = assign({ count: (_, e) => e.count })
setCount.displayName = 'setCount'
```

This will also deal with code modification if the inspector is exposed for production builds.

**Comment by Andarist** [comment]
Shouldn't the name just be sourced from the "options"/"implementations" object (the second one passed to the `createMachine`)? We wouldn't have to implement any special handling for `.displayName` with that in place.

**Reply by VanTanev** [reply]
Yes, but some people prefer to use functions defined at the top level. It might be nice to accommodate them, but it is definitely something that's not a strong need.

**Reply by Andarist** [reply]
I think this might be a good topic for the later point in time - we should first expand on the visual builder (which might steer people towards "providing" implementations using the second argument) and think about introducing things like that only if there will still be a demand for such a thing then.
